sled-agent/src/sim/storage.rs

```diff
@@ -1581,7 +1597,11 @@ impl StorageInner {
                 self.next_crucible_port + 100,
             );
 
+        // Due to how the sled index works, there's a limit of 10 crucible
+        // agents per dataset: they're separated by 1000 in
+        // [`StorageInner::new`]. Bail here if that limit is reached.
         self.next_crucible_port += 100;
+        assert!(self.next_crucible_port < 1000);
```

Comment:
I'm missing something here. In `new()` I see

```
next_crucible_port: (sled_index + 1) * 1000,
```

which looks like this assertion would always fail (since `next_crucible_port` starts at N-thousand for the Nth sled-agent)?

Similarly confused question: the comment says "10 per dataset", but `next_crucible_port` looks like it's incremented independently of datasets. Is it 10 per dataset or 10 total?

Reply:
I think both of us are missing something here! This worked for me when I ran the full test suite before opening this PR, but now I'm seeing

```
thread 'test_omdb_env_settings' panicked at sled-agent/src/sim/storage.rs:1604:9:
assertion failed: self.next_crucible_port < 1000
```

What should be asserted is the next port - starting port is less than the range, which is done in d440dc0b5. That commit also changes the ranges so that there can be a maximum of 20 datasets/agents per sled (and therefore 50 regions/snapshots per agent) and also clarifies the comment.

Reply:
Ahh, that makes sense. Thanks, the new comment is very helpful.